--- dosfstools-3.0.1.orig/debian/docs
+++ dosfstools-3.0.1/debian/docs
@@ -0,0 +1 @@
+debian/doc/*
--- dosfstools-3.0.1.orig/debian/rules
+++ dosfstools-3.0.1/debian/rules
@@ -0,0 +1,46 @@
+#!/usr/bin/make -f
+
+clean:
+	dh_testdir
+	dh_testroot
+	rm -f build-stamp
+
+	$(MAKE) distclean
+
+	dh_clean
+
+build: build-stamp
+build-stamp:
+	dh_testdir
+
+	$(MAKE)
+
+	touch build-stamp
+
+install: build
+	dh_testdir
+	dh_testroot
+	dh_prep
+	dh_installdirs
+
+	$(MAKE) DESTDIR=$(CURDIR)/debian/dosfstools PREFIX=/usr SBINDIR=/sbin install
+
+binary: binary-arch
+
+binary-arch: build install
+	dh_testdir
+	dh_testroot
+	dh_installchangelogs ChangeLog
+	dh_installdocs
+	dh_strip --dbg-package=dosfstools-dbg
+	dh_compress
+	dh_fixperms
+	dh_installdeb
+	dh_shlibdeps
+	dh_gencontrol
+	dh_md5sums
+	dh_builddeb
+
+binary-indep:
+
+.PHONY: clean build install binary binary-arch binary-indep
--- dosfstools-3.0.1.orig/debian/control
+++ dosfstools-3.0.1/debian/control
@@ -0,0 +1,35 @@
+Source: dosfstools
+Section: otherosfs
+Priority: optional
+Maintainer: Daniel Baumann <daniel@debian.org>
+Build-Depends: debhelper (>= 7)
+Standards-Version: 3.8.0
+Homepage: http://www.daniel-baumann.ch/software/dosfstools/
+Vcs-Browser: http://git.debian.net/?p=debian/dosfstools.git
+Vcs-Git: git://git.debian.net/git/debian/dosfstools.git
+
+Package: dosfstools
+Architecture: any
+Depends: ${shlibs:Depends}, ${misc:Depends}
+Description: utilities for making and checking MS-DOS FAT filesystems
+ The dosfstools package includes the mkdosfs (aka mkfs.dos and mkfs.vfat) and
+ dosfsck (aka fsck.msdos and fsck.vfat) utilities, which respectively make and
+ check MS-DOS FAT filesystems on hard drives or on floppies.
+ .
+ This version uses the enhanced boot sector/superblock format of DOS 3.3+ as
+ well as provides a default dummy boot sector code.
+
+Package: dosfstools-dbg
+Section: devel
+Priority: extra
+Architecture: any
+Depends: dosfstools (= ${binary:Version})
+Description: utilities for making and checking MS-DOS FAT filesystems (debug)
+ The dosfstools package includes the mkdosfs (aka mkfs.dos and mkfs.vfat) and
+ dosfsck (aka fsck.msdos and fsck.vfat) utilities, which respectively make and
+ check MS-DOS FAT filesystems on hard drives or on floppies.
+ .
+ This version uses the enhanced boot sector/superblock format of DOS 3.3+ as
+ well as provides a default dummy boot sector code.
+ .
+ This package contains the debugging symbols.
--- dosfstools-3.0.1.orig/debian/changelog
+++ dosfstools-3.0.1/debian/changelog
@@ -0,0 +1,427 @@
+dosfstools (3.0.1-1) unstable; urgency=low
+
+  * Replacing obsolete dh_clean -k with dh_prep.
+  * Merging upstream version 3.0.1:
+    - Changing some wording to make the indended meaning of "full-disk device"
+      more obvious (Closes: #500623).
+
+ -- Daniel Baumann <daniel@debian.org>  Sun, 23 Nov 2008 22:51:46 +0100
+
+dosfstools (3.0.0-1) unstable; urgency=low
+
+  * Merging upstream version 3.0.0.
+  * Updating homepage field in control.
+  * Updating upstream information in copyright file.
+  * Removing now useless lintian override.
+  * Removing now useless manpage.
+  * Removing now useless patches.
+  * Removing now useless docs.
+  * Updating rules to for new upstream.
+
+ -- Daniel Baumann <daniel@debian.org>  Sun, 28 Sep 2008 11:56:00 +0200
+
+dosfstools (2.11-8) unstable; urgency=low
+
+  * Adding debug package.
+
+ -- Daniel Baumann <daniel@debian.org>  Sat, 20 Sep 2008 21:52:00 +0200
+
+dosfstools (2.11-7) unstable; urgency=low
+
+  * Adding manpage for dosfslabel, thanks to Francois Wendling
+    <frwendling@free.fr> (Closes: #496760).
+  * Updating vcs fields in control file.
+  * Using patch-stamp rather than patch in rules file.
+
+ -- Daniel Baumann <daniel@debian.org>  Sat, 20 Sep 2008 21:48:00 +0200
+
+dosfstools (2.11-6) unstable; urgency=high
+
+  * Temporarily disabling bootcode.dpatch since it breaks syslinux,
+    thanks to Joey Hess <joeyh@debian.org> (Closes: #489292).
+
+ -- Daniel Baumann <daniel@debian.org>  Fri,  4 Jul 2008 20:28:00 +0200
+
+dosfstools (2.11-5) unstable; urgency=medium
+
+  * Adding patch to mention fsckNNNN.rec files in dosfsck manpage
+    (Closes: #444596).
+  * Adding patch from Sam Bingner <sam@bingner.com> to add option for using a
+    bootcode template (Closes: #303442).
+  * Adding patch from Adonikam Virgo <adonikam@virgonet.org> to fix backup
+    sector getopt (Closes: #232387, #479794).
+  * Adding patch from Karl Tomlinson <karlt@karlt.net> to fix segfaults with
+    zero slots in lfn (Closes: #152550, #353198, #356377, #401798).
+  * Rediffing 99-conglomeration.dpatch.
+  * Adding patch from Eero Tamminen <eero.tamminen@nokia.com> to improve memory
+    efficiencey when checking filesystems.
+  * Adding patch to list alternative binary names in manpage synopsis
+    (Closes: #284983).
+  * Adding patch to not deny FAT32 auto-selection in mkdosfs manpage
+    (Closes: #414183).
+  * Adding patch to not use confusing 'drop' in dosfsck manpage where 'delete'
+    in dosfsck manpage where 'delete' is meant (Closes: #134100).
+  * Listing alternative binary names in long-description (Closes: #434381).
+  * Updating manpage spelling patch to also cover wrong acknowledge header
+    (Closes: #306659).
+  * Breaking out manpage typos patch.
+
+ -- Daniel Baumann <daniel@debian.org>  Fri, 27 Jun 2008 09:03:00 +0200
+
+dosfstools (2.11-4) unstable; urgency=low
+
+  * Redone debian packaging from scratch.
+  * Both stop avoiding -O2 and stop adding -fno-strict-aliasing to OPTFLAGS on
+    alpha; seems not to be required anymore.
+  * Added patch from Jakub Jelinek <jakub@redhat.com> to support
+    -D_FORTIFY_SOURCE=2 (for future use).
+  * Added patch from Jeremy Katz <katzj@redhat.com> to add dosfslabel
+    (originally by Peter Jones).
+  * Added patch from Pavol Rusnak <prusnak@suse.cz> to use O_EXCL in mkdosfs.
+  * Added patch from Petr Gajdos <pgajdos@suse.cz> to automatically determine
+    sector size of the device.
+
+ -- Daniel Baumann <daniel@debian.org>  Thu, 26 Jun 2008 13:13:00 +0200
+
+dosfstools (2.11-3) unstable; urgency=low
+
+  * New maintainer (Closes: #488018).
+
+ -- Daniel Baumann <daniel@debian.org>  Thu, 26 Jun 2008 12:31:00 +0200
+
+dosfstools (2.11-2.3) unstable; urgency=low
+
+  * NMU
+  * dpkg --print-gnu-build-architecture is gone, use dpkg-architecture
+    instead. Closes: #407192
+  * Fixed errors in the mkdosfs manpage that could cause confusion to
+    readers, Thanks to Onno Benschop. Closes: #433561 (LP: #126121)
+
+ -- Joey Hess <joeyh@debian.org>  Mon, 10 Sep 2007 15:57:36 -0400
+
+dosfstools (2.11-2.2) unstable; urgency=low
+
+  * Non-maintainer upload.
+  * Apply Ubuntu patches. Closes: #417673
+
+ -- Andreas Barth <aba@not.so.argh.org>  Sat,  9 Jun 2007 21:18:21 +0000
+
+dosfstools (2.11-2.1) unstable; urgency=low
+
+  * Non-maintainer upload.
+  * Fix unaligned memory accesses which cause warnings to appear everytime
+    the elilo bootloader script runs.  This has led a number of users to
+    believe their install has failed.  Thanks to Khalid Aziz for the patch.
+    Closes: #258839.
+
+ -- dann frazier <dannf@debian.org>  Thu,  9 Jun 2005 18:14:45 -0600
+
+dosfstools (2.11-2) unstable; urgency=low
+
+  * Oops, debian/rules overrides OPTFLAGS and therefore the
+    -D_FILE_OFFSET_BITS=64 in the toplevel Makefile had no effect; added
+    $(shell getconf LFS_CFLAGS) to OPTFLAGS as suggested by Lars Wirzenius in
+    #300126. Sorry, I tested a version compiled by the upstream Makefile...
+    Closes: #300126, #301254.
+  * #302517 was indeed the same as #294177, and fix is the same (use __u8) as
+    in 2.11-1. Closes: #302517.
+
+ -- Roman Hodek <roman@hodek.net>  Sun,  3 Apr 2005 13:56:55 +0200
+
+dosfstools (2.11-1) unstable; urgency=low
+
+  * New upstream version (Closes: #293394, #295181, #294177, #270023, #258402,
+    #232482, #214656, #286219, #276834, #266254, #128800)
+
+ -- Roman Hodek <roman@hodek.net>  Sat, 12 Mar 2005 17:19:27 +0100
+
+dosfstools (2.10-1) unstable; urgency=low
+
+  * New upstream version:
+     - dosfsck: various 64-bit fixes and removed some warnings by Michal
+       Cihar <mcihar@suse.cz>
+     - mkdosfs: better error message if called without parameters (also
+       suggested by Michal)
+  * recompilation removed errno@GLIBC_2.0 symbol; Closes: #168540
+    (probably already 2.9-1 did)
+
+ -- Roman Hodek <roman@hodek.net>  Mon, 22 Sep 2003 22:15:32 +0200
+
+dosfstools (2.9-1) unstable; urgency=low
+
+  * New upstream version:
+   (Closes: #156266, #139198, #152769, #152868, #181196)
+    - dosfsck: if EOF from stdin, exit with error code
+   	- dosfsck: Fix potential for "Internal error: next_cluster on bad cluster".
+   	- dosfsck: When clearing long file names, don't overwrite the dir
+   	  entries with all zeros, but put 0xe5 into the first byte.
+   	  Otherwise, some OSes stop reading the directory at that point...
+   	- dosfsck: in statistics printed by -v, fix 32bit overflow in number
+   	  of data bytes.
+   	- dosfsck: fix an potential overflow in "too many clusters" check
+   	- dosfsck: fix 64bit problem in fat.c (Debian bug #152769)
+   	- dosfsck: allow FAT size > 32MB.
+   	- dosfsck: allow for only one FAT
+   	- dosfsck: with -v, also check that last sector of the filesystem can
+   	  be read (in case a partition is smaller than the fs thinks)
+   	- mkdosfs: add note in manpage that creating bootable filesystems is
+   	  not supported.
+   	- mkdosfs: better error message with pointer to -I if target is a
+   	  full-disk device.
+  * debian/control: Added build dependency on debhelper (Closes: #168388)
+  * debian/control: spelling fix (Closes: #124564)
+  * debian/control: metion names of tools in description (Closes: #186047)
+
+ -- Roman Hodek <roman@hodek.net>  Thu, 15 May 2003 20:54:04 +0200
+
+dosfstools (2.8-1) unstable; urgency=low
+
+  * New upstream version fixing an endless loop.
+    (Closes: #87205, #86373, #87590)
+
+ -- Roman Hodek <roman@hodek.net>  Wed, 28 Feb 2001 17:23:16 +0100
+
+dosfstools (2.7-1) unstable; urgency=low
+
+  * New upstream version with various bug fixes. (Closes: #83883)
+  * Changed maintainer e-mail addr.
+
+ -- Roman Hodek <roman@hodek.net>  Wed, 14 Feb 2001 12:49:00 +0100
+
+dosfstools (2.6-1) unstable; urgency=low
+
+  * New upstream version with various bug fixes.
+
+ -- Roman Hodek <roman@hodek.net>  Tue, 28 Nov 2000 17:27:35 +0100
+
+dosfstools (2.5-1) unstable; urgency=low
+
+  * New upstream version fixing llseek() on alpha (Closes: #54145)
+
+ -- Roman Hodek <Roman.Hodek@informatik.uni-erlangen.de>  Fri, 7 Jan 2000 09:26:51 +0100
+
+dosfstools (2.4-1) unstable; urgency=low
+
+  * New upstream version fixing compilation problem on alpha (Closes: #48331)
+
+ -- Roman Hodek <Roman.Hodek@informatik.uni-erlangen.de>  Tue, 26 Oct 1999 09:38:39 +0200
+
+dosfstools (2.3-1) unstable; urgency=low
+
+  * New upstream version:
+     - Manpage fixes.
+     - Fixed usage message of mkdosfs.
+     - Fixed compilation on alpha (llseek).
+     - Fixed unaligned accesses on alpha (Closes: #47714)
+     - Fixed renaming of files in dosfsck (extension wasn't really
+       written). Closes: #45774
+  * Remove bashisms from debian/rules.
+  * FHS transition.
+  * On alpha, omit -O2 and add -fno-strict-aliasing to OPTFLAGS, as
+    otherwise the programs fail with unaligned traps.
+  * FHS transition (Standards-Version 3.0.1).
+
+ -- Roman Hodek <Roman.Hodek@informatik.uni-erlangen.de>  Fri, 22 Oct 1999 13:30:59 +0200
+
+dosfstools (2.2-1) unstable; urgency=low
+
+  * New upstream version. Closes: #40533
+  * Updated copyright file for new location /usr/share/common-licenses/GPL.
+  * Updated Standards-Version to 3.0.0.
+
+ -- Roman Hodek <Roman.Hodek@informatik.uni-erlangen.de>  Tue, 6 Jul 1999 16:07:22 +0200
+
+dosfstools (2.1-1) unstable; urgency=low
+
+  * New upstream version.
+  * Also installs symlinks mkfs.vfat and fsck.vfat,so that also
+    filesystems listed with type "vfat" in /etc/fstab can be automatically
+    checked.
+
+ -- Roman Hodek <Roman.Hodek@informatik.uni-erlangen.de>  Fri, 28 May 1999 11:25:17 +0200
+
+dosfstools (2.0-1) unstable; urgency=low
+
+  * I'm now upstream maintainer, too.
+  * Adapted debian/rules for new central Makefile and rewritten to
+    debhelper.
+  * Fixed harmless warnings in mkdosfs.c and fat.c.
+  * mkdosfs.c: Return type of getopt() must be stored in an int, not in a
+    char. (Showed up on powerpc, where chars are default unsigned.)
+
+ -- Roman Hodek <Roman.Hodek@informatik.uni-erlangen.de>  Fri, 30 Apr 1999 14:37:37 +0200
+
+dosfstools (1.0-16) unstable; urgency=low
+
+  * Both tools are not maintained upstreams anymore, so I take over
+    general maintainership now...
+  * Both mkdosfs and dosfsck now support the FAT32 format. dosfsck
+    automatically detects it. For mkdosfs, 32 is now a valid argument to
+    -F, but FAT32 format isn't selected automatically (yet). With FAT32,
+    you can also set the number of reserved sectors (-R) and location of
+    the boot sector backup (-b) if you like (defaults are reasonable).
+  * dosfsck can handle and check VFAT-style long filenames now. It uses
+    the long names in listing etc. when available. There are also some
+    checks on the structures of LFNs and some fixes for possible problems.
+  * Implemented Atari format in both, dosfsck and mkdosfs. Under Atari TOS
+    basically the same FAT format is used as under DOS, however, there are
+    some little differences. Both tools now automatically select Atari
+    format if they run on an Atari. You can switch between standard MS-DOS
+    and Atari format with the -A option.
+  * Applied patch by Giuliano Procida <gpp10@cus.cam.ac.uk> to add loop
+    device support to mkdosfs: Usual floppy sizes are detected and
+    parameters (media byte,...) are set up accordingly.
+    My own additions to this: Don't die on loop devices that don't have
+    such a floppy size, but use some default hd params. Added endianess
+    conversions to Giulianos patch.
+  * More/better data in boot sector dump of dosfsck -v.
+  * Fixed lots of gcc warnings in the source. Removed -Wno-parentheses flag.
+  * Made dosfsck -v a bit more verbose.
+  * Extended README's for FAT32/LFN.
+  * Written a README for Atari format differences.
+  * Some minor cleanups in debian/rules.
+  * Install README files as README.{mkdosfs,dosfsck,Atari}.gz.
+  * AFAIK the tools still have alignment problems on Alpha machines.
+    Someone wanted to send me a patch, but I haven't heard from him for
+    months...
+  * Set Standards-Version to 2.5.0.0 (no changes).
+
+ -- Roman Hodek <Roman.Hodek@informatik.uni-erlangen.de>  Wed, 28 Apr 1999 11:06:15 +0200
+
+dosfstools (1.0-15) frozen unstable; urgency=low
+
+  * Applied patch by Juan Cespedes <cespedes@debian.org> to make mkdosfs
+    work with newer 2.1 kernels again. (Fixes: #20320)
+  * Remove CC=gcc in debian/rules to make cross-compiling possible.
+
+ -- Roman Hodek <Roman.Hodek@informatik.uni-erlangen.de>  Thu, 30 Apr 1998 17:09:08 +0200
+
+dosfstools (1.0-14) frozen unstable; urgency=medium
+
+  * New maintainer
+  * Ignore long name directory slots of VFAT, instead of trying to correct
+    that "file names". Fixes: #20711
+  * Don't consider file names with chars >= 128 to be bad, they're allowed.
+
+ -- Roman Hodek <Roman.Hodek@informatik.uni-erlangen.de>  Thu, 30 Apr 1998 10:00:16 +0200
+
+dosfstools (1.0-13) stable; urgency=low
+
+  * Fixed nasty bug that caused every file with a name like xxxxxxxx.xxx
+    to be treated as bad name that needed to be fixed. (closes: Bug#17389)
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Thu, 26 Feb 1998 01:14:40 +0100
+
+dosfstools (1.0-12) stable; urgency=low
+
+  * Moved executables and their links into /sbin (Bug#15037)
+  * Corrected Standards-Version to 2.3.0.1
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Fri, 9 Jan 1998 21:49:48 +0100
+
+dosfstools (1.0-11) stable; urgency=low
+
+  * Applied patches to source files from Juan Cespedes which got lost in
+    the last upload (Bug#16493, Bug#16494, Bug#16490)
+  * build-stamp is now removed first in clean target (Bug#16491)
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Sat, 3 Jan 1998 15:06:27 +0100
+
+dosfstools (1.0-10) stable; urgency=low
+
+  * Added links for mkfs.msdos and fsck.msdos (Bug#15037)
+  * Added links for mkfs.msdos.8 and fsck.msdos.8
+  * Corrected source location
+  * Rewrote bad_name() (Bug#9871, part 2)
+  * s/int/time_t in check.c to omit a compiler warning (Bug#9871, part 3.1)
+  * Modified defaults, interactive is the default now
+  * Fixed renaming of files, they are saved now (Bug#9871, part 1)
+  * Fixed return type of date_dos2unix (Bug#9871, part 3.2)
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Wed, 31 Dec 1997 18:59:53 +0100
+
+dosfstools (1.0-9.1) unstable; urgency=low
+
+  * Non-maintainer release
+  * .diff file was wrong in 1.0-9; fixed (Bug#13102)
+  * Fixed endianess patches (fixes Bug#11648)
+  * Built with libc6
+
+ -- Juan Cespedes <cespedes@etsit.upm.es>  Thu, 23 Oct 1997 23:19:34 +0200
+
+dosfstools (1.0-9) stable; urgency=low
+
+  * Added endianess patches from Frank Neumann
+    <Frank.Neumann@Informatik.Uni-Oldenburg.DE> (Bug#9959)
+  * Updated Werner Almesbergers address
+  * Added -I switch to mkdosfs to allow full disk devices (Bug#10789)
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Sun, 20 Jul 1997 14:51:33 +0200
+
+dosfstools (1.0-8) stable; urgency=low
+
+  * Fixed typo in mkdosfs.c (Bug#7396)
+
+  * New maintainer address
+
+ -- Martin Schulze <joey@infodrom.north.de>  Mon, 28 Apr 1997 12:51:13 +0200
+
+dosfstools (1.0-7) stable; urgency=low
+
+  * Minor fixes
+
+  * Converted to Standards-Version 2.1.1.2
+
+ -- Martin Schulze <joey@debian.org>  Sat, 8 Feb 1997 15:03:52 +0100
+
+Mon Jan 20 22:45:23 1997  Martin Schulze  <joey@finlandia.infodrom.north.de>
+
+	* debian.control: Corrected otherfs to otherosfs :-)
+
+Sun Jan 19 23:57:09 1997  Martin Schulze  <joey@finlandia.infodrom.north.de>
+
+	* mkdosfs.c: Removed one line printing out debug information.
+
+Wed Jan 15 00:25:02 1997  Martin Schulze  <joey@finlandia.infodrom.north.de>
+
+	* mkdosfs.c: Modified code to work properly under AXP. Thanks to
+	Matt Gundry <mjgundry@primenet.com> for contacting me and
+	providing patched sources.
+	
+Mon Jan 13 13:00:14 1997  Martin Schulze  <joey@finlandia.infodrom.north.de>
+
+	* mkdosfs.c: Added patch from Sven Rudolph <sr1@inf.tu-dresden.de>
+ 	to support creation on disk images as well.
+
+Tue Jan  7 12:04:21 1997  Martin Schulze  <joey@finlandia.infodrom.north.de>
+
+	* Moved into section 'otherfs'
+
+-- Released 1.0-4
+	
+Sun Dec 22 11:28:03 1996  Martin Schulze  <joey@finlandia.infodrom.north.de>
+
+	* debian.rules: Installed ChangeLog
+
+Wed Aug  7 19:07:15 1996  Martin Schulze  <joey@finlandia.infodrom.north.de>
+
+	* Corrected debian.rules to provide $(package)_$(revision).diff.gz
+
+Mon Aug  5 11:13:34 1996  Martin Schulze  <joey@finlandia.infodrom.north.de>
+
+	* Added Conflicts: and Provides: entries in control file to really
+	replace mkdosfs.  Thanks to Michael Meskes for reporting this bug.
+
+Thu Jul 18 22:01:34 1996  Martin Schulze  <joey@finlandia.infodrom.north.de>
+
+	* added dosfscheck to the package
+
+Wed Jul  3 00:57:23 1996  Martin Schulze  <joey@finlandia.infodrom.north.de>
+
+	* Added debian specific files
+
+	* mkdosfs.c: inserted some brackets to stop gcc from moaning
+
+	* mkdosfs.8: corrected some bold/inverse text phrases
+
+	* mkdosfs.c: initialized a variable to stop gcc from moaning
--- dosfstools-3.0.1.orig/debian/copyright
+++ dosfstools-3.0.1/debian/copyright
@@ -0,0 +1,88 @@
+Author: Daniel Baumann <daniel@debian.org>
+Download: http://www.daniel-baumann.ch/software/dosfstools/
+
+Files: *
+Copyright: (C) 1999-2005 Roman Hodek <Roman.Hodek@informatik.uni-erlangen.de>
+License: GPL-2+
+ This program is free software; you can redistribute it and/or
+ modify it under the terms of the GNU General Public License
+ as published by the Free Software Foundation; either version 2
+ of the License, or (at your option) any later version.
+ .
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ GNU General Public License for more details.
+ .
+ You should have received a copy of the GNU General Public License
+ along with this program; if not, write to the Free Software
+ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ .
+ On Debian systems, the complete text of the GNU General Public License
+ can be found in /usr/share/common-licenses/GPL-2 file.
+
+Files: dosfsck/*
+Copyright:
+ (C) 1999-2005 Werner Almesberger <werner.almesberger@lrc.di.epfl.ch>
+ (C) 1999-2005 Roman Hodek <Roman.Hodek@informatik.uni-erlangen.de>
+License: GPL-2+
+ This program is free software; you can redistribute it and/or
+ modify it under the terms of the GNU General Public License
+ as published by the Free Software Foundation; either version 2
+ of the License, or (at your option) any later version.
+ .
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ GNU General Public License for more details.
+ .
+ You should have received a copy of the GNU General Public License
+ along with this program; if not, write to the Free Software
+ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ .
+ On Debian systems, the complete text of the GNU General Public License
+ can be found in /usr/share/common-licenses/GPL-2 file.
+
+Files: mkdosfs/*
+Copyright:
+ (C) 1993-1994 David Hudson <dave@humbug.demon.co.uk>
+ (C) 1991 Linus Torvalds <torvalds@klaava.helsinki.fi>
+ (C) 1992-1993 Remy Card <card@masi.ibp.fr>
+ (C) 1999-2005 Roman Hodek <Roman.Hodek@informatik.uni-erlangen.de>
+License: GPL-2+
+ This program is free software; you can redistribute it and/or
+ modify it under the terms of the GNU General Public License
+ as published by the Free Software Foundation; either version 2
+ of the License, or (at your option) any later version.
+ .
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ GNU General Public License for more details.
+ .
+ You should have received a copy of the GNU General Public License
+ along with this program; if not, write to the Free Software
+ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ .
+ On Debian systems, the complete text of the GNU General Public License
+ can be found in /usr/share/common-licenses/GPL-2 file.
+
+Files: debian/*
+Copyright: (C) 2008 Daniel Baumann <daniel@debian.org>
+License: GPL-2+
+ This program is free software; you can redistribute it and/or
+ modify it under the terms of the GNU General Public License
+ as published by the Free Software Foundation; either version 2
+ of the License, or (at your option) any later version.
+ .
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ GNU General Public License for more details.
+ .
+ You should have received a copy of the GNU General Public License
+ along with this program; if not, write to the Free Software
+ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ .
+ On Debian systems, the complete text of the GNU General Public License
+ can be found in /usr/share/common-licenses/GPL-2 file.
--- dosfstools-3.0.1.orig/debian/compat
+++ dosfstools-3.0.1/debian/compat
@@ -0,0 +1 @@
+7
--- dosfstools-3.0.1.orig/debian/patches/14-bootcode.dpatch
+++ dosfstools-3.0.1/debian/patches/14-bootcode.dpatch
@@ -0,0 +1,235 @@
+#!/bin/sh /usr/share/dpatch/dpatch-run
+## 14-bootcode.dpatch by Sam Bingner <sam@bingner.com>
+##
+## DP: Adds option for using a bootcode template (Closes: #303442).
+
+@DPATCH@
+
+diff -Naurp dosfstools.orig/mkdosfs/ChangeLog dosfstools/mkdosfs/ChangeLog
+--- dosfstools.orig/mkdosfs/ChangeLog	2008-06-27 05:37:31.000000000 +0000
++++ dosfstools/mkdosfs/ChangeLog	2008-06-27 06:44:05.000000000 +0000
+@@ -1,3 +1,14 @@
++19th June 2003			Sam Bingner (sam@bingner.com)
++
++	Added option to read in bootcode from a file so that if you have
++	for example Windows 2000 boot code, you can have it write that
++	as the bootcode.  This is a dump of the behinning of a partition
++	generally 512 bytes, but can be up to reserved sectors*512 bytes.
++	Also writes 0x80 as the BIOS drive number if we are formatting a
++	hard drive, and sets the number of hidden sectors to be the
++	number of sectors in one track. These were required so that DOS
++	could boot using the bootcode.
++
+ 28th January 1995		H. Peter Anvin (hpa@yggdrasil.com)
+ 
+ 	Better algorithm to select cluster sizes on large filesystems.
+diff -Naurp dosfstools.orig/mkdosfs/mkdosfs.8 dosfstools/mkdosfs/mkdosfs.8
+--- dosfstools.orig/mkdosfs/mkdosfs.8	2008-06-27 06:34:35.000000000 +0000
++++ dosfstools/mkdosfs/mkdosfs.8	2008-06-27 06:44:05.000000000 +0000
+@@ -44,6 +44,10 @@
+ .I message-file
+ ]
+ [
++.B \-B
++.I bootcode-file
++]
++[
+ .B \-n
+ .I volume-name
+ ]
+@@ -163,6 +167,18 @@ file must not exceed 418 bytes once line
+ carriage return-line feed combinations, and tabs have been expanded.
+ If the filename is a hyphen (-), the text is taken from standard input. 
+ .TP
++.BI \-B " bootcode-file"
++Uses boot machine code from file "file".  On any thing other than FAT32,
++this only writes the first 3 bytes, and 480 bytes from offset 3Eh.  On
++FAT32, this writes the first 3 bytes, 420 bytes from offset 5Ah to both
++primary and backup boot sectors.  Also writes all other reserved sectors
++excluding the sectors following boot sectors (usually sector 2 and 7).
++Does not require that the input file be as large as reserved_sectors*512.
++To make a FAT32 partition bootable, you will need at least the first
++13 sectors (6656 bytes).  You can also specify a partition as the argument
++to clone the boot code from that partition.
++i.e mkdosfs -B /dev/sda1 /dev/sda1
++.TP
+ .BI \-n " volume-name"
+ Sets the volume name (label) of the file system.  The volume name can
+ be up to 11 characters long.  The default is no label.
+diff -Naurp dosfstools.orig/mkdosfs/mkdosfs.c dosfstools/mkdosfs/mkdosfs.c
+--- dosfstools.orig/mkdosfs/mkdosfs.c	2008-06-27 06:34:35.000000000 +0000
++++ dosfstools/mkdosfs/mkdosfs.c	2008-06-27 06:50:19.000000000 +0000
+@@ -24,6 +24,12 @@
+    - New options -A, -S, -C
+    - Support for filesystems > 2GB
+    - FAT32 support
++
++   Fixes/additions June 2003 by Sam Bingner
++   <sam@bingner.com>:
++   - Add -B option to read in bootcode from a file
++   - Write BIOS drive number so that FS can properly boot
++   - Set number of hidden sectors before boot code to be one track
+    
+    Copying:     Copyright 1993, 1994 David Hudson (dave@humbug.demon.co.uk)
+ 
+@@ -153,6 +159,8 @@ cdiv (int a, int b)
+ #define FAT_BAD      0x0ffffff7
+ 
+ #define MSDOS_EXT_SIGN 0x29	/* extended boot sector signature */
++#define HD_DRIVE_NUMBER 0x80	/* Boot off first hard drive */
++#define FD_DRIVE_NUMBER 0x00	/* Boot off first floppy drive */
+ #define MSDOS_FAT12_SIGN "FAT12   "	/* FAT12 filesystem signature */
+ #define MSDOS_FAT16_SIGN "FAT16   "	/* FAT16 filesystem signature */
+ #define MSDOS_FAT32_SIGN "FAT32   "	/* FAT32 filesystem signature */
+@@ -175,6 +183,8 @@ cdiv (int a, int b)
+ #define BOOTCODE_SIZE		448
+ #define BOOTCODE_FAT32_SIZE	420
+ 
++#define MAX_RESERVED		0xFFFF
++
+ /* __attribute__ ((packed)) is used on all structures to make gcc ignore any
+  * alignments */
+ 
+@@ -202,7 +212,7 @@ struct msdos_boot_sector
+   __u16         fat_length;	/* sectors/FAT */
+   __u16         secs_track;	/* sectors per track */
+   __u16         heads;		/* number of heads */
+-  __u32         hidden;		/* hidden sectors (unused) */
++  __u32         hidden;		/* hidden sectors (one track) */
+   __u32         total_sect;	/* number of sectors (if sectors == 0) */
+   union {
+     struct {
+@@ -285,6 +295,8 @@ char dummy_boot_code[BOOTCODE_SIZE] =
+ 
+ /* Global variables - the root of all evil :-) - see these and weep! */
+ 
++static char *template_boot_code;	/* Variable to store a full template boot sector in */
++static int use_template = 0;
+ static char *program_name = "mkdosfs";	/* Name of the program */
+ static char *device_name = NULL;	/* Name of the device on which to create the filesystem */
+ static int atari_format = 0;	/* Use Atari variation of MS-DOS FS format */
+@@ -837,6 +849,12 @@ setup_tables (void)
+     vi->volume_id[2] = (unsigned char) ((volume_id & 0x00ff0000) >> 16);
+     vi->volume_id[3] = (unsigned char) (volume_id >> 24);
+   }
++  if (bs.media == 0xf8) {
++      vi->drive_number = HD_DRIVE_NUMBER;  /* Set bios drive number to 80h */
++  }
++  else {
++      vi->drive_number = FD_DRIVE_NUMBER;  /* Set bios drive number to 00h */
++  }
+ 
+   if (!atari_format) {
+     memcpy(vi->volume_label, volume_name, 11);
+@@ -881,7 +899,7 @@ setup_tables (void)
+     printf( "Using %d reserved sectors\n", reserved_sectors );
+   bs.fats = (char) nr_fats;
+   if (!atari_format || size_fat == 32)
+-    bs.hidden = CT_LE_L(hidden_sectors);
++    bs.hidden = bs.secs_track;
+   else {
+     /* In Atari format, hidden is a 16 bit field */
+     __u16 hidden = CT_LE_W(hidden_sectors);
+@@ -1362,6 +1380,32 @@ write_tables (void)
+    * dir area on FAT12/16, and the first cluster on FAT32. */
+   writebuf( (char *) root_dir, size_root_dir, "root directory" );
+ 
++  if (use_template == 1) {
++    /* dupe template into reserved sectors */
++    seekto( 0, "Start of partition" );
++    if (size_fat == 32) {
++      writebuf( template_boot_code, 3, "backup jmpBoot" );
++      seekto( 0x5a, "sector 1 boot area" );
++      writebuf( template_boot_code+0x5a, 420, "sector 1 boot area" );
++      seekto( 512*2, "third sector" );
++      if (backup_boot != 0) {
++        writebuf( template_boot_code+512*2, backup_boot*sector_size - 512*2, "data to backup boot" );
++	seekto( backup_boot*sector_size, "backup boot sector" );
++        writebuf( template_boot_code, 3, "backup jmpBoot" );
++	seekto( backup_boot*sector_size+0x5a, "backup boot sector boot area" );
++        writebuf( template_boot_code+0x5a, 420, "backup boot sector boot area" );
++        seekto( (backup_boot+2)*sector_size, "sector following backup code" );
++        writebuf( template_boot_code+(backup_boot+2)*sector_size, (reserved_sectors-backup_boot-2)*512, "remaining data" );
++      } else {
++        writebuf( template_boot_code+512*2, (reserved_sectors-2)*512, "remaining data" );
++      }
++    } else {
++      writebuf( template_boot_code, 3, "jmpBoot" );
++      seekto( 0x3e, "sector 1 boot area" );
++      writebuf( template_boot_code+0x3e, 448, "boot code" );
++    }
++  }
++
+   if (blank_sector) free( blank_sector );
+   if (info_sector) free( info_sector );
+   free (root_dir);   /* Free up the root directory space from setup_tables */
+@@ -1376,7 +1420,7 @@ usage (void)
+ {
+   fatal_error("\
+ Usage: mkdosfs [-A] [-c] [-C] [-v] [-I] [-l bad-block-file] [-b backup-boot-sector]\n\
+-       [-m boot-msg-file] [-n volume-name] [-i volume-id]\n\
++       [-m boot-msg-file] [-n volume-name] [-i volume-id] [-B bootcode]\n\
+        [-s sectors-per-cluster] [-S logical-sector-size] [-f number-of-FATs]\n\
+        [-h hidden-sectors] [-F fat-size] [-r root-dir-entries] [-R reserved-sectors]\n\
+        /dev/name [blocks]\n");
+@@ -1440,7 +1484,7 @@ main (int argc, char **argv)
+   printf ("%s " VERSION " (" VERSION_DATE ")\n",
+ 	   program_name);
+ 
+-  while ((c = getopt (argc, argv, "Ab:cCf:F:Ii:l:m:n:r:R:s:S:h:v")) != EOF)
++  while ((c = getopt (argc, argv, "AB:b:cCf:F:Ii:l:m:n:r:R:s:S:h:v")) != EOF)
+     /* Scan the command line for options */
+     switch (c)
+       {
+@@ -1510,6 +1554,51 @@ main (int argc, char **argv)
+ 	listfile = optarg;
+ 	break;
+ 
++      case 'B':         /* B : read in bootcode */
++        if ( strcmp(optarg, "-") )
++	  {
++	    msgfile = fopen(optarg, "r");
++	    if ( !msgfile )
++	      perror(optarg);
++	  }
++	else
++	  msgfile = stdin;
++
++	if ( msgfile )
++	  {
++            if (!(template_boot_code = malloc( MAX_RESERVED )))
++                die( "Out of memory" );
++	    /* The template boot sector including reserved must not be > 65535 */
++            use_template = 1;
++	    i = 0;
++	    do
++	      {
++		ch = getc(msgfile);
++		switch (ch)
++		  {
++		  case EOF:
++		    break;
++
++		  default:
++		    template_boot_code[i++] = ch; /* Store character */
++		    break;
++		  }
++	      }
++	    while ( ch != EOF && i < MAX_RESERVED );
++	    ch = getc(msgfile); /* find out if we're at EOF */
++
++	    /* Fill up with zeros */
++	    while( i < MAX_RESERVED )
++		template_boot_code[i++] = '\0';
++
++	    if ( ch != EOF )
++	      printf ("Warning: template too long; truncated after %d bytes\n", i);
++
++	    if ( msgfile != stdin )
++	      fclose(msgfile);
++	  }
++	break;
++
+       case 'm':		/* m : Set boot message */
+ 	if ( strcmp(optarg, "-") )
+ 	  {
--- dosfstools-3.0.1.orig/debian/doc/HISTORY
+++ dosfstools-3.0.1/debian/doc/HISTORY
@@ -0,0 +1,20 @@
+This package was put together by Martin Schulze <joey@infodrom.north.de>, from
+sources obtained from:
+  sunsite.unc.edu:/pub/Linux/system/filesystems/dosfs/mkdosfs-ygg-0.3b.lsm
+  sunsite.unc.edu:/pub/Linux/system/filesystems/dosfs/mkdosfs-ygg-0.3b.tar.gz
+  sunsite.unc.edu:/pub/Linux/system/filesystems/dosfs/dosfsck.1.tar.z
+
+Mkdosfs is copyrighted by Dave Hudson <dave@humbug.demon.co.uk>, and was later
+maintained and improved by H. Peter Anvin <hpa@yggdrasil.com>. Dave has put the
+whole package under the GPL.
+
+Dosfsck is copyrighted by Werner Almesberger <werner.almesberger@lrc.di.epfl.ch>
+and Roman Hodek. We both together put the package under the GPL.
+
+The FAT32, LFN, and Atari format additions are copyrighted by Roman Hodek
+<Roman.Hodek@informatik.uni-erlangen.de> and are under the GPL, too.
+
+Both tools (mkdosfs and dosfsck) have been unmaintained since some time, so I
+also took them over as upstream maintainer. The merged package is named
+dosfstools-2.0, and its home site is now
+  ftp.uni-erlangen.de:/pub/Linux/LOCAL/dosfstools
